tests/src/astro/electromagnetism/unitTestYarkovskyAcceleration.cpp

```diff
@@ -0,0 +1,207 @@
+/*    Copyright (c) 2010-2019, Delft University of Technology
+ *    All rights reserved
+ *
+ *    This file is part of the Tudat. Redistribution and use in source and
+ *    binary forms, with or without modification, are permitted exclusively
+ *    under the terms of the Modified BSD license. You should have received
+ *    a copy of the license with this file. If not, please or visit:
+ *    http://tudat.tudelft.nl/LICENSE.
+ */
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MAIN
+
+#include <boost/test/unit_test.hpp>
+#include <Eigen/Core>
+
+#include "tudat/basics/testMacros.h"
+#include "tudat/astro/electromagnetism/yarkovskyAcceleration.h"
+#include "tudat/astro/basic_astro/physicalConstants.h"
+
+#include "tudat/math/basic/linearAlgebra.h"
+#include "tudat/astro/basic_astro/orbitalElementConversions.h"
+#include "tudat/interface/spice/spiceInterface.h"
+#include "tudat/math/integrators/rungeKuttaCoefficients.h"
+#include "tudat/astro/basic_astro/accelerationModel.h"
+#include "tudat/simulation/propagation_setup/dynamicsSimulator.h"
+#include "tudat/simulation/environment_setup/defaultBodies.h"
+#include "tudat/simulation/environment_setup/createBodies.h"
+#include "tudat/math/integrators/createNumericalIntegrator.h"
+
+
+namespace tudat
+{
+
+namespace unit_tests
+{
+
+BOOST_AUTO_TEST_SUITE( test_yarkovsky_acceleration )
+
+const double AU = physical_constants::ASTRONOMICAL_UNIT;
+const double JD = physical_constants::JULIAN_DAY;
+const double yarkovskyParameter = -2.899e-14 * AU / ( JD * JD );
+
+//! Test the implementation of the Yarkovsky Acceleration Model
+BOOST_AUTO_TEST_CASE( testYarkovskyAccelerationVerySimple )
+{
+    const Eigen::Vector6d state = { 0.5 * AU, 0.0, 0.0, 0.0, 5e4, 0.0 };
+
+    const Eigen::Vector3d computedYarkovskyAcceleration = electromagnetism::computeYarkovskyAcceleration(
+            yarkovskyParameter,
+            state );
+    const Eigen::Vector3d expectedYarkovskyAcceleration = 4 * Eigen::Vector3d { 0.0, yarkovskyParameter, 0.0
+    };
+
+    TUDAT_CHECK_MATRIX_CLOSE( computedYarkovskyAcceleration, expectedYarkovskyAcceleration, 1.e-10 );
+}
+
+//! Test the implementation of the Yarkovsky Acceleration Model
+BOOST_AUTO_TEST_CASE( testYarkovskyAccelerationSimple )
+{
+    const Eigen::Vector6d state = { 2.0 * AU, AU, 1.0e10, 3.0e5, 4.0e5, 12.0e5
+    };
+
+    const Eigen::Vector3d expectedYarkovskyAcceleration =
+            yarkovskyParameter * 0.19982142476807888 * Eigen::Vector3d { 3.0, 4.0, 12.0 } / 13.0;
```

Comment:
Where do the numbers in this line come from?

Reply:
The numbers are arbitrary. I just chose a convenient vector to take the norm for. What would be a better approach?

Reply:
Having arbitrary numbers is perfectly fine, my main doubt is where the very specific 0.19982142476807888 value comes from